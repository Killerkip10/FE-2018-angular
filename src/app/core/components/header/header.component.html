<mat-toolbar color="primary">
  <a [routerLink]="['/']">{{'HEADER.LOGO' | translate}}</a>

  <div>
    <mat-select #select (selectionChange)="selectLanguage($event)" [value]="translateService.getLangs()[0]">
      <mat-option *ngFor="let lang of translateService.getLangs()" [value]="lang">{{lang}}</mat-option>
    </mat-select>

    <button mat-flat-button color="primary" (click)="logout()">{{'SIGN-OUT' | translate}}</button>
  </div>
</mat-toolbar>
