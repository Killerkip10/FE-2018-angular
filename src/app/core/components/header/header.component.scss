mat-toolbar{
  display: flex;
  justify-content: space-between;
  height: 56px;
  >:last-child{
    display: flex;
    align-items: center;
    mat-select{
      display: block;
      width: 50px;
      margin-right: 10px;
    }
  }
}
