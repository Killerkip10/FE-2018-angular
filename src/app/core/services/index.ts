export * from './auth.service';
export * from './profile.service';
export * from './token.service';
export * from './search-user.service';
