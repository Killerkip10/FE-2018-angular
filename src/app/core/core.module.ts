import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {TranslateModule} from '@ngx-translate/core';
import {HttpClientModule} from '@angular/common/http';
import {RouterModule} from '@angular/router';

import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
import {AuthGuard} from './guards';
import {AuthService} from './services';
import {
  MatButtonModule,
  MatToolbarModule,
  MatSelectModule,
  MatIconModule
} from '@angular/material';
import {
  HeaderComponent,
  FooterComponent
} from './components';

@NgModule({
  imports: [
    BrowserModule,
    HttpClientModule,
    TranslateModule,
    BrowserAnimationsModule,

    RouterModule,

    MatIconModule,
    MatButtonModule,
    MatSelectModule,
    MatToolbarModule
  ],
  declarations: [
    HeaderComponent,
    FooterComponent
  ],
  providers: [
    AuthService,

    AuthGuard
  ],
  exports: [
    HeaderComponent,
    FooterComponent
  ]
})
export class CoreModule {}
