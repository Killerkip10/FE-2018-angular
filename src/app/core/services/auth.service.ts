import {Injectable} from '@angular/core';
import {Observable, throwError} from 'rxjs';
import {catchError, map} from 'rxjs/internal/operators';
import {HttpErrorResponse, HttpResponse} from '@angular/common/http';
import {Router} from '@angular/router';
import {HttpClient} from '@angular/common/http';

import {urlConfig, options} from '../../config';
import {Login} from '../../models';
import {TokenService} from './token.service';
import {ProfileService} from './profile.service';

@Injectable()
export class AuthService {
  constructor(
    private userService: ProfileService,
    private router: Router,
    private http: HttpClient
  ) {}

  public login(loginObj: Login) {
    return this.http.post(urlConfig.login, loginObj, options)
      .pipe(
        map(resp => this.loginHelper(resp)),
        catchError(err => this.handleLoginError(err))
      );
  }
  public forgotPassword(login: string): Observable<string> {
    return this.http.get<HttpResponse<string>>(urlConfig.forgot + login, options)
      .pipe(
        map(resp => resp.body),
        catchError(err => this.handleForgotError(err))
      );
  }
  public logout(): void {
    TokenService.deleteToken();
    this.userService.deleteUser();
    this.router.navigate(['/login']);
  }

  private handleLoginError(error: HttpErrorResponse) {
    return throwError(error.status === 401 ? 'ERROR.LOGIN' : 'ERROR.BASE');
  }
  private handleForgotError(error: HttpErrorResponse) {
    return throwError(error.status === 404 ? 'ERROR.FORGOT' : 'ERROR.BASE');
  }
  private loginHelper(resp): void {
    this.userService.updateUser();
    this.router.navigate(['/']);
  }
}
