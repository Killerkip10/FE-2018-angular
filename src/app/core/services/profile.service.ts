import {Injectable} from '@angular/core';
import {BehaviorSubject} from 'rxjs';
import {map} from 'rxjs/internal/operators';
import {HttpClient, HttpResponse} from '@angular/common/http';

import {User} from '../../models/index';
import {urlConfig, options} from '../../config';
import {TokenService} from './token.service';

@Injectable()
export class ProfileService {
  private user = new BehaviorSubject<User>(null);

  constructor(private http: HttpClient) {
    this.updateUser();
  }

  public get getUserSubject(): BehaviorSubject<User> {
    return this.user;
  }

  public updateUser(): void {
    if (TokenService.getToken()) {
      this.http.get<HttpResponse<User>>(urlConfig.getProfile, options)
        .subscribe(
          resp => this.user.next(resp.body)
        );
    }
  }
  public deleteUser(): void {
    this.user.next(null);
  }
  public editUser(user: User) {
    return this.http.put<HttpResponse<User>>(urlConfig.editProfile, user, options)
      .pipe(
        map(resp => {
          this.user.next(resp.body);
          return resp;
        })
      );
  }
  public checkName(name: string) {
    return this.http.get(urlConfig.checkName + name, options);
  }
}
