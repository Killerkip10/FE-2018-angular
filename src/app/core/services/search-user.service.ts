import {Injectable} from '@angular/core';
import {HttpClient, HttpResponse, HttpParams, HttpHeaders} from '@angular/common/http';
import {Observable} from 'rxjs';
import {map} from 'rxjs/operators';

import {urlConfig} from '../../config';
import {User} from '../../models';

@Injectable()
export class SearchUserService {
  constructor(private http: HttpClient) { }

  public findUser(name: string): Observable<User[]> {
    const options = <object>{
      headers: new HttpHeaders({
        'content-type': 'application/json'
      }),
      observe: 'response',
      withCredentials: true,
      params: new HttpParams().set('name', name)
    };
    return this.http.get<HttpResponse<User[]>>(urlConfig.findUsers, options)
      .pipe(
        map(resp => resp.body)
      );
  }
}
