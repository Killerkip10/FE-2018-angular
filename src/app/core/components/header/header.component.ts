import {Component} from '@angular/core';
import {TranslateService} from '@ngx-translate/core';
import {Router} from '@angular/router';

import {Store} from '@ngrx/store';
import {DeleteProfile} from '../../../store/actions/profile';

import {AuthService} from '../../services';

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.scss']
})
export class HeaderComponent {
  constructor(
    public translateService: TranslateService,
    private authService: AuthService,
    private router: Router,
    private store: Store<null>
  ) {}

  public selectLanguage(event): void {
    this.translateService.use(event.value);
  }
  public logout(): void {
    this.authService.logout();
    this.store.dispatch(new DeleteProfile());
    this.router.navigate(['/login']);
  }
}
