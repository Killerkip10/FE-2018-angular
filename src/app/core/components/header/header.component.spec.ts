import {NO_ERRORS_SCHEMA} from '@angular/core';
import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {TranslateModule} from '@ngx-translate/core';
import {HttpClientModule} from '@angular/common/http';
import {APP_BASE_HREF, CommonModule} from '@angular/common';
import {RouterTestingModule} from '@angular/router/testing';

import {StoreModule} from '@ngrx/store';
import {EffectsModule} from '@ngrx/effects';
import {reducers} from '../../../store/reducers';
import {effects} from '../../../store/effects';

import {HeaderComponent} from './header.component';
import {AppTranslate} from '../../../app.translate';
import {AuthService} from '../../services';


describe('HeaderComponent', () => {
  let component: HeaderComponent;
  let fixture: ComponentFixture<HeaderComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        HttpClientModule,
        TranslateModule,
        RouterTestingModule,
        CommonModule,

        StoreModule.forRoot(reducers),
        EffectsModule.forRoot(effects),

        AppTranslate
      ],
      declarations: [
        HeaderComponent
      ],
      providers: [
        AuthService,
        {provide: APP_BASE_HREF, useValue: '/'}
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
