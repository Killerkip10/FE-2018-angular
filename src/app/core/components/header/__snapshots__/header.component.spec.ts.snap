// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent snapshot 1`] = `
<app-header
  authService={[Function AuthService]}
  router={[Function Router]}
  store={[Function Store]}
  translateService={[Function TranslateService]}
>
  <mat-toolbar
    color="primary"
  >
    <a
      href="/"
      ng-reflect-router-link="/"
    >
      HEADER.LOGO
    </a>
    <div>
      <mat-select>
        
      </mat-select>
      <button
        color="primary"
        mat-flat-button=""
      >
        SIGN-OUT
      </button>
    </div>
  </mat-toolbar>
</app-header>
`;
